site/themes/sogilis/static/js/scripts.js

```diff
@@ -121,3 +120,200 @@ const throwError = (message) => {
+    return response.json();
+  }
+
+  return [];
+};
+
+/**
+ * @property {HTMLLIElement} getElementsByTagName
+ */
+class SearchBarResultItem {
+  /**
+   * @param {string} postTitle
+   * @param {string} postUrl
+   * @param {RegExp} regexp
+   */
+  constructor(postTitle, postUrl, regexp) {
+    const li = document.createElement('li');
+    li.classList.add('search-bar__results-item');
+
+    const link = document.createElement('a');
+    link.classList.add('search-bar__results-item-link');
+    link.href = postUrl;
+    link.innerHTML = postTitle;
+
+    li.append(link);
+
+    this.element = li;
+    this.element.firstElementChild.innerHTML = this.element.firstElementChild.innerHTML.replace(
+      regexp,
+      (match) => `<mark>${match}</mark>`
+    );
+  }
+}
+
+/**
+ * @property {HTMLFormElement}  form
+ * @property {HTMLLabelElement}      label
+ * @property {HTMLInputElement} input
+ * @property {HTMLUListElement} searchResults
+ */
+class SearchBar extends HTMLElement {
+  constructor() {
+    super();
+
+    this.shortcutHandler = this.shortcutHandler.bind(this);
+    this.toggleInput = this.toggleInput.bind(this);
+  }
+
+  connectedCallback() {
+    this.innerHTML = `
+      <form id="search" class="search-bar" role="search">
+        <label for="search-input" class="search-bar__label" id="search-bar-label">
+          <?xml version="1.0" encoding="iso-8859-1"?>
+          <svg class="search-bar__icon" version="1.1" id="Capa_1" xmlns="http://www.w3.org/2000/svg" xmlns:xlink="http://www.w3.org/1999/xlink" x="0px" y="0px"
+            viewBox="0 0 512.005 512.005" style="enable-background:new 0 0 512.005 512.005;" xml:space="preserve">
+          <g>
+            <g>
+              <path d="M505.749,475.587l-145.6-145.6c28.203-34.837,45.184-79.104,45.184-127.317c0-111.744-90.923-202.667-202.667-202.667
+                S0,90.925,0,202.669s90.923,202.667,202.667,202.667c48.213,0,92.48-16.981,127.317-45.184l145.6,145.6
+                c4.16,4.16,9.621,6.251,15.083,6.251s10.923-2.091,15.083-6.251C514.091,497.411,514.091,483.928,505.749,475.587z
+                M202.667,362.669c-88.235,0-160-71.765-160-160s71.765-160,160-160s160,71.765,160,160S290.901,362.669,202.667,362.669z"/>
+            </g>
+          </g>
+          <g></g><g></g><g></g><g></g><g></g><g></g><g></g><g></g><g></g><g></g><g></g><g></g><g></g><g></g><g></g>
+          </svg>
+        </label>
+        <input type="search" id="search-input" class="search-bar__input" placeholder="Search...">
+      </form>
+      <div class="search-bar__results" hidden><ul id="search-result"></ul></div>
+    `;
+
+    this.form = this.querySelector('#search');
+    this.label = this.querySelector('#search-bar-label');
+    this.input = this.querySelector('#search-input');
+    this.searchResults = this.querySelector('#search-result');
+
+    this.noResultElement = document.createElement('h1');
+    this.noResultElement.innerHTML = 'No results found.';
```

Comment:
```suggestion
    this.noResultElement.innerHTML = 'Aucun résultat trouvé.
```

Vu que l'interface du blog est en français.